### PR TITLE
Add candy-machine instructions to update authority and close config account

### PR DESCRIPTION
**Pull request by lwus**
The candy machine state is not closed since it is a small account. @bhgames

### CODE REVIEW

File: rust/nft-candy-machine/src/lib.rs [hunk 1]
```diff
@@ -499,6 +518,27 @@ pub struct UpdateCandyMachine<'info> {
...
+    #[account(
+        mut,
+        has_one = config,
+        has_one = authority,
+        seeds = [PREFIX.as_bytes(), config.key().as_ref(), candy_machine.data.uuid.as_bytes()],
+        bump = candy_machine.bump,
+    )]
+    candy_machine: ProgramAccount<'info, CandyMachine>,
+    #[account(signer)]
+    authority: AccountInfo<'info>,
```
Comment (bhgames) [inline rust/nft-candy-machine/src/lib.rs:537]: Use Signer not account(signer)

**Comment by lwus** [comment]
Closing to defer to earlier https://github.com/metaplex-foundation/metaplex/pull/857/commits/8a17fc7cb2d5d70f9d1b2cc5b0ccd5dbd9950b5a. Update auth now in separate PR https://github.com/metaplex-foundation/metaplex/pull/876

**Comment by brandonculver** [comment]
> Closing to defer to earlier [8a17fc7](https://github.com/metaplex-foundation/metaplex/commit/8a17fc7cb2d5d70f9d1b2cc5b0ccd5dbd9950b5a). Update auth now in separate PR #876

The PR you are referring to was closed and I haven't seen an new one opened to close out unsold candy machines. Also your solution offered a charity option. It seems like its a better solution. Unless there is communication happening privately around this I would create PR for your solution.

**Comment by lwus** [comment]
@brandonculver sorry bad link w.r.t config refunds: https://github.com/metaplex-foundation/metaplex/pull/857.

I believe @staccDOTsol is working with @bhgames to get charity et. al. working.

**Pull request closed without merging**